ci/test.sh

```
@@ -1,101 +1,159 @@
+		SSH_FINGERPRINT=$(ssh-keygen -F '[localhost]:2222' -f "${HOME}/.ssh/known_hosts" -l | tail -n 1 | cut -d ' ' -f 2 | tr -d ':')
+	else
+		SSH_FINGERPRINT=$(ssh-keygen -E md5 -F '[localhost]:2222' -f "${HOME}/.ssh/known_hosts" -l | tail -n 1 | cut -d ' ' -f 3 | cut -d : -f2- | tr -d :)
+	fi
 fi
 
 # Run the tests that do not require network connectivity.
 
-echo ""
-echo "################################################################################"
-echo "## Running (non-online) tests                                                 ##"
-echo "################################################################################"
+if [ -z "$SKIP_OFFLINE_TESTS" ]; then
+	echo ""
+	echo "##############################################################################"
+	echo "## Running (offline) tests"
+	echo "##############################################################################"
 
-#ctest -V -R offline
+	ctest -V -R offline || die $?
+fi
 
-# Run the various online tests.  The "online" test suite only includes the
-# default online tests that do not require additional configuration.  The
-# "proxy" and "ssh" test suites require further setup.
+if [ -z "$SKIP_ONLINE_TESTS" ]; then
+	# Run the various online tests.  The "online" test suite only includes the
+	# default online tests that do not require additional configuration.  The
+	# "proxy" and "ssh" test suites require further setup.
 
-echo ""
-echo "################################################################################"
-echo "## Running (online) tests                                                     ##"
-echo "################################################################################"
+	echo ""
+	echo "##############################################################################"
+	echo "## Running (online) tests"
+	echo "##############################################################################"
 
-#ctest -V -R online
+	ctest -V -R online || die $?
```

Do we really want to die if any of those tests fails? I think it would be more sensible to just record that a test has failed and die after _all_ tests ran. If you die eagerly, you'll potentially have to re-run CI every time you have fixed a test just to stumble into the next one that was previously not executed.

---

This is what we're doing today.  🤷‍♂️ I would rather have us continue to fail fast, on the hopes that we get out of the CI queue as quickly as possible.  I guess that could lead to someone spamming the queue with brokenness, but we haven't seen that so far.